### PR TITLE
Handle pie executables on Linux

### LINKED ISSUE #51: Handle PIE executables on Linux by reading mem maps

### PR DESCRIPTION
**Pull request by bjorn3**
Fixes #51

### CODE REVIEW

File: src/target/unix.rs [hunk 1]
```diff
@@ -32,6 +33,12 @@ pub(crate) fn launch(path: &str) -> Result<Pid, Box<dyn std::error::Error>> {
         ForkResult::Child => {
             ptrace::traceme()?;
 
+            // Disable ASLR
+            #[cfg(target_os = "linux")]
+            unsafe {
+                libc::personality(0x0040000 /*ADDR_NO_RANDOMIZE*/);
```
Comment (nbaksalyar) [inline src/target/unix.rs:39]: Nit: this probably would be better as
```suggestion
               const ADDR_NO_RANDOMIZE: usize = 0x0040000;
               libc::personality(ADDR_NO_RANDOMIZE);
```
It also seems that this const is missing in rust-lang/libc, I'm going to raise a PR for that :)

File: src/target/mod.rs [hunk 2]
```diff
@@ -12,3 +12,11 @@ pub use linux::*;
 mod macos;
 #[cfg(target_os = "macos")]
 pub use macos::*;
+
+#[derive(Debug)]
+pub struct MemoryMap {
```
Comment (nbaksalyar) [inline src/target/mod.rs:17]: Nice one, this is also helpful for the `macos` target :+1:

File: src/symbol/mod.rs [hunk 3]
```diff
@@ -277,3 +280,132 @@ impl Dwarf {
         self.rent(|parsed| parsed.get_var_address(name))
     }
 }
+
+pub struct RelocatedDwarf(Vec<RelocatedDwarfEntry>);
```
Comment (nbaksalyar) [inline src/symbol/mod.rs:284]: This struct probably can be moved into its own module, but can be done later.

File: src/target/unix.rs (hunk 1 above)
Comment (bjorn3) [reply src/target/unix.rs:39]: Yeah, all personalities are missing. I had to `cat /usr/include/sys/personality.h` to find the correct value.

File: src/symbol/mod.rs (hunk 3 above)
Comment (bjorn3) [reply src/symbol/mod.rs:284]: Will do this later.

**Comment by nbaksalyar** [comment]
> It also seems that this const is missing in rust-lang/libc, I'm going to raise a PR for that :)

Raised https://github.com/rust-lang/libc/pull/1849

**Pull request merged**